frontend/sw360-portlet/src/main/webapp/js/modules/codeScoop.js

```diff
@@ -0,0 +1,966 @@
+
+        this._init_missed_release_button = function (size, url) {
+            var releaseButton = document.createElement("button");
+            releaseButton.innerHTML = "Populate Releases (" + size + ")";
+            releaseButton.setAttribute("class","addButton");
+            releaseButton.style.cursor = "pointer";
+
+            var target = document
+                .getElementsByClassName("pageHeader")[0]
+                .getElementsByClassName("pull-right")[0];
+            target.insertBefore(releaseButton, target.firstChild);
+
+            releaseButton.onclick = function() {
+                window.location = url;
+            };
+        };
+
+        this._clean_string_array = function (array) {
+            var result = [];
+            for (var i = 0; i < array.length; i++) {
+                var str = array[i];
+                str = str.trim();
+                if (str.length > 0) {
+                    result.push(str);
+                }
+            }
+            return result;
+        };
+
+        this._match_string_array = function (array1, array2) {
+            if (array1.length !== array2.length) {
+                return false;
+            }
+            for (var i = 0; i < array1.length; i++) {
+                if (array1[i] !== array2[i]) {
+                    return false;
+                }
+            }
+            return true;
+        };
+
+        this._check_component_diff = function (sw360Component, externalComponent) {
+            var result = {};
+
+            if (sw360Component.description !== externalComponent.title) {
+                result["description"] = externalComponent.title;
+            }
+
+            if (externalComponent.homepageUrl === "") {
+                externalComponent.homepageUrl = externalComponent.url;
+            }
+
+            if (sw360Component.homepage !== externalComponent.homepageUrl) {
+                result["homepage"] = externalComponent.homepageUrl;
+            }
+
+            if (!_this._match_string_array(sw360Component.categories, externalComponent.categories)) {
+                result["categories"] = externalComponent.categories;
+            }
+
+            if (!_this._match_string_array(sw360Component.languages, externalComponent.langs)) {
+                result["languages"] = externalComponent.langs;
+            }
+
+            if (sw360Component.licenses.indexOf(externalComponent.license) < 0) {
+                sw360Component.licenses.push(externalComponent.license);
+                result["licenses"] = sw360Component.licenses;
+            }
+            return result;
+        };
+
+        this.activateAutoFill = function() {
+            _this._install_form_element();
+            _this._install_autocomplete_box();
+            _this._listen_autocomplete();
+        };
+
+        this.activateIndexes = function(tableID, dataUrl) {
+            dataUrl = dataUrl + "&sEcho=1&iColumns=5&sColumns=%2C%2C%2C%2C&iDisplayStart=0" +
```

Comment:
This long hardcoded string looks fragile

Reply:
@maxhbr this is part constant for all kind request...
is it better solution to make this works?

Reply:
one solution could using the builder pattern to build an URL. i am not sure if this works straight away in javascript though. For java there are some implementations for this.

Reply:
@mcjaeger unfortunately I can not pass to such build data from JS